src/pubtools/_marketplacesvm/tasks/push/command.py

```diff
@@ -212,12 +212,14 @@ def _push_upload(
             # Upload the VM image to the marketplace
             # In order to get the correct destinations we need to first pass the result of
             # get_push_item_from_marketplace.
+            pi = mapped_item.get_push_item_for_marketplace(marketplace)
+            meta = mapped_item.get_metadata_for_mapped_item(pi.dest[0]) or {}
```

Comment:
Why do we choose `dest[0]` ?
We can have more destinations e.g. rhcos-x86_64-aws

Reply:
I thought because the metadata for each might be the same, but yes, it may lead to some issues.

I'll work on fixing that.

Reply:
Please check the commit 3ba3eaf1830a10d89607d16b19951e6bf16e574f

I've added a comment there, stating that we expect the "same" sharing accounts for each destination in the same marketplace.

This is different from the community, because there each region may have different settings and we do lots of uploads (each for a different region). Here, however, we do a single upload for a given marketplace, no matter how many offers it has as destinations, the `sharing_accounts` should be the same.

With that I opted to merge the meta dictionary for that property so it has the same effect as it had in the past. It's something very specific for the marketplaces case, which we don't upload to different regions and we don't need different settings for each upload.

For the publishing it's retrieving the information correctly.